lib/mix/lib/mix/release.ex

```diff
@@ -297,30 +298,44 @@ defmodule Mix.Release do
     end
   end
 
-  defp load_app(app, seen, otp_root, included) do
-    path = Path.join(otp_root, "#{app}-*")
+  defp load_app(app, apps_lock, seen, otp_root, included) do
+    case apps_lock[app] do
+      {:hex, _, version, _, _, _, _, _} ->
```

Comment:
I could not find all possibilities for the lock entry

Reply:
Yeah, unfortunately the lock format is opaque, so you cannot match on it. :/ However, I don't think you need to know the lock entry itself, you simply need to know if it is a Mix dependency or not!

So my suggestion is to simply call `Mix.Project.deps_apps`. If the app is in `deps_apps`, then you don't look at it in the otp_root, only in lib_dir!

Reply:
Ok, I gave a shot to this then. I had understood first that this would conflict with https://github.com/elixir-lang/elixir/pull/9054 🤔 
The setup for this function to work in the test is quite convoluted, maybe there is some more straightforward manner